cmd/restic/cmd_dump.go

```diff
@@ -90,13 +90,31 @@ func printFromTree(ctx context.Context, tree *restic.Tree, repo restic.Repositor
 				return printFromTree(ctx, subtree, repo, item, pathComponents[1:], pathToPrint)
 			case node.Type == "dir":
 				node.Path = pathToPrint
-				return tarTree(ctx, repo, node, pathToPrint)
+				tw := tar.NewWriter(os.Stdout)
+				defer tw.Close()
```

Comment:
You're dropping an error returned by `tw.Close()`, which needs to be checked. I'd remove this line and handle it as shown below.
```suggestion
```